activerecord/test/cases/create_new_entry.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require "bundler/inline"
+require "cases/helper"
+require "active_record"
+require "minitest/autorun"
+require "logger"
+
+ActiveRecord::Base.establish_connection(adapter: "sqlite3", database: ":memory:")
+ActiveRecord::Base.logger = Logger.new(STDOUT)
+
+ActiveRecord::Schema.define do
+  create_table :posts, force: true do |t|
+  end
+end
+
+class CreateNewEntryTest < ActiveRecord::TestCase
+  class Post < ActiveRecord::Base
+    class << self
+      def prefetch_primary_key?
+        true
+      end
+
+      def next_sequence_value
+        Random.rand(1000000)
+      end
+    end
+  end
+
+  class BugTest < Minitest::Test
+    def test_prefetch_primary_key
+      Post.create!
+      assert true
```

Comment:
While it's understood that by reaching `assert true` line without failing we imply that `Post.create!` was successful and no exception was raised. But generally I find an explicit assertion like `assert_nothing_raised { Post.create! }` to be even more reader friendly.

However, what I find even more friendly is presence of a clear assertion. In our case we expect that `next_sequence_value` is what will define the primary key value for us, so perhaps let's add an explicit expectation by making `next_sequence_value` constant and asserting for the actual value?

For example:

```ruby
def next_sequence_value
  123_456
end

def test_create_prefetches_primary_key_value
  post = assert_nothing_raised { Post.create! }
  assert_equal 123_456, post.id
end
```

We can use one of the existing AR test models, `Post` for example - https://github.com/rails/rails/blob/f5a1075e71b79eb7eb0f4f6323468ef9c7286ee9/activerecord/test/models/post.rb#L3
but we could define a subclass which defines the prefetch behavior:

```ruby
  class PostWithPrefetchedPk < Post
    class << self
      def prefetch_primary_key?
        true
      end

      def next_sequence_value
        Random.rand(1000000)
      end
    end
  end
```

Or perhaps it's unnecessary to inherit from the existing one and our post can be a standalone model like
```ruby
  class PostWithPrefetchedPk < ActiveRecord::Base
    class << self
      def prefetch_primary_key?
        true
      end

      def next_sequence_value
        Random.rand(1000000)
      end
    end
  end
end
```
but with a unique name to avoid conflicts

Reply:
Great suggestions! The test can definitely be made more meaningful.

Reply:
After doing some further research, it appears the assert_nothing_raised is deprecated (source: https://apidock.com/ruby/Test/Unit/Assertions/assert_nothing_raised).

Would it be okay to just add an explicit expectation for next_sequence_value?

Reply:
It's okay to omit `assert_nothing_raised` since we will have a much better explicit assertion for the primary key value but just wanted to mention that it must be a wrong `apidock` link since `assert_nothing_raised` at least the one defined in Rails itself is not deprecated and available
https://api.rubyonrails.org/classes/ActiveSupport/Testing/Assertions.html#method-i-assert_nothing_raised

https://github.com/rails/rails/blob/23938052acd773fa24068debe56cd892cbf8d868/activesupport/lib/active_support/testing/assertions.rb#L48-L52

Reply:
Thank you for the reply. What the article is saying is strange because I see assert_nothing_raised in other parts of the Rails codebase.